internal/error.go

```diff
@@ -0,0 +1,80 @@
+package fleetlock
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// ErrorKind is a error type identifier.
+//
+// It is designed to be tracked in Zincati metrics.
+// Fleetlock specification states that this MUST have a bounded/small cardinality.
+type ErrorKind string
+
+// List of ErrorKinds
+const (
+	ErrorMethodNotAllowed ErrorKind = "method_not_allowed"
+	ErrorMissingHeader    ErrorKind = "missing_header"
+	ErrorDecodingRequest  ErrorKind = "failed_decoding_request"
+	ErrorInternal         ErrorKind = "internal_error"
+	ErrorLocked           ErrorKind = "locked"
+)
+
+// Error holds data used for http responses in case of error.
+type Error struct {
+	// error type identifier
+	Kind ErrorKind `json:"kind"`
+
+	// human-friendly error description
+	Message string `json:"value"`
+
+	StatusCode int `json:"-"`
```

Comment:
Handle HTTP details at response writing, rather than in the message type itself.